site/src/request_handlers/github.rs

```diff
@@ -25,96 +27,197 @@ pub async fn handle_github(
     ctxt: Arc<SiteCtxt>,
 ) -> ServerResult<github::Response> {
     log::info!("handle_github({:?})", request);
-    if request.comment.body.contains(" homu: ") {
-        if let Some(sha) = parse_homu_comment(&request.comment.body).await {
-            enqueue_sha(request.issue, &ctxt, sha).await?;
+    match request {
+        github::Request::Issue { issue, comment } => handle_issue(ctxt, issue, comment).await,
+        github::Request::Push(p) => handle_push(ctxt, p).await,
+    }
+}
+
+async fn handle_push(ctxt: Arc<SiteCtxt>, push: github::Push) -> ServerResult<github::Response> {
+    let client = reqwest::Client::new();
+    let repository_url = "https://github.com/rust-lang-ci/rust";
+    if push.r#ref != "refs/heads/master"
+        || !is_rollup(&client, &ctxt, repository_url, &push).await?
+    {
+        return Ok(github::Response);
+    }
+
+    let previous_master = &push.before;
+    let rollup_merges = push
+        .commits
+        .iter()
+        .rev()
+        .skip(1) // skip the head commit
+        .take_while(|c| c.message.starts_with("Rollup merge of "))
```

Comment:
I think this probably should be a filter - the commits don't have a single linear ordering due to the merges, so I don't think we can rely on it being the first (or last) N.

Did you test this perhaps?

Reply:
I believe this is correct. The commits seem to always be (in reverse order) - the merge commit for the entire roll up, each individual merge commit for the rolled up PRs, and then the individual commits.

Reply:
Ok, seems okay for now. Can always adjust in the future.